Robot.java_03_14_17

```diff
@@ -1,418 +1 @@
-		RightButton.whenPressed(new SetTankDriveBack());
-		
-		HighButton.whenPressed(new FrontCamera());
-		LowButton.whenPressed(new BackCamera());
-		//TopButton.whenPressed(new GreenLight());
-		
-		//RightButton.whenPressed(new MoveForward());
-		
-		
-		state = "forward1";
-		
-	}
-
-	
-	
-	/**
-	 * This function is called once each time the robot enters Disabled mode.
-	 * You can use it to reset any subsystem information you want to clear when
-	 * the robot is disabled.
-	 */
-	@Override
-	public void disabledInit() {
-		this.left.disable();
-		this.right.disable();
-
-	}
-
-	@Override
-	public void disabledPeriodic() {
-		Scheduler.getInstance().run();
-	}
-
-	/**
-	 * This autonomous (along with the chooser code above) shows how to select
-	 * between different autonomous modes using the dashboard. The sendable
-	 * chooser code works with the Java SmartDashboard. If you prefer the
-	 * LabVIEW Dashboard, remove all of the chooser code and uncomment the
-	 * getString code to get the auto name from the text box below the Gyro
-	 *
-	 * You can add additional auto modes by adding additional commands to the
-	 * chooser code above (like the commented example) or additional comparisons
-	 * to the switch structure below with additional strings & commands.
-	 */
-	@Override
-	public void autonomousInit() {
-		autonomousCommand = chooser.getSelected();
-		
-		
-		double KpR=9/72.0;
-		double KpL=9/72.0;
-		double Ki = 0.01/72.0;
-		double Kd= 1/72.0;
-		//double KfR= 1.4/36.0;
-		//double KfL=1.45/36.0;
-		double Kf = 1.0;
-		
-		this.right = new Good_DriveSide(KpR, Ki, Kd, Kf, rightEncoder, rightSpark);
-		right.setInverted(true);
-		this.left = new Good_DriveSide(KpL, Ki, Kd, Kf, leftEncoder, leftSpark);
-
-		
-		driveTrain = new RobotDrive(left,right);
-
-		/*
-		 * String autoSelected = SmartDashboard.getString("Auto Selector",
-		 * "Default"); switch(autoSelected) { case "My Auto": autonomousCommand
-		 * = new MyAutoCommand(); break; case "Default Auto": default:
-		 * autonomousCommand = new ExampleCommand(); break; }
-		 */
-		//autonomousCommand = new MoveForward();
-		// schedule the autonomous command (example)
-		if (autonomousCommand != null)
-			autonomousCommand.start();
-		
-
-		myTimer.reset();
-		myTimer.start();
-		
-		
-		System.out.println("Autonomous!");
-		System.out.println("StartTime: " + myTimer.get());
-		System.out.println();
-		System.out.println();
-		
-	//leftEncoder.reset();
-		gyro.calibrate();
-//		System.out.println("gyro calibration: " + gyro.getAngle());
-		
-		
-		
-	}
-
-	/**
-	 * This function is called periodically during autonomous
-	 */
-	@Override
-	public void autonomousPeriodic() {
-		
-		Scheduler.getInstance().run();
-	
-		//	System.out.println("encoder count: "+ ((Good_DriveSide) left).getEncoderCount());
-			//QUESTION: if i made my own method inside good_driveside, which implements speedController
-			//do I have to cast left as a goodDriveside instead of speedcontroller?
```

Comment:
Yes, if you want to access those functions, you have to cast it to GoodDriveside